File: SDK/Python/allbeapi.py
import requests
import json

class AllBeApi:
    """
    AllBeAPI Python SDK - Minimal Utility Function Set
    Base URL: https://res.allbeapi.top

    Rapid development toolkit for prototyping and experimentation.
    Access ready-to-use functions without installing heavy dependencies.
    """
    def __init__(self, base_url='https://res.allbeapi.top'):
        """
        Initializes a new instance of the AllBeApi client.
        :param base_url: The base URL for the API.
        """
        self.base_url = base_url
        self.session = requests.Session()
        self.session.headers.update({'Content-Type': 'application/json'})

        self.marked = MarkedAPI(self)
        self.beautifulsoup = BeautifulSoupAPI(self)
        self.prettier = PrettierAPI(self)
        self.pygments = PygmentsAPI(self)
        self.python_qrcode = PythonQRCodeAPI(self) # python-qrcode
        self.sanitize_html = SanitizeHtmlAPI(self) # sanitize-html
        self.ajv = AjvAPI(self)
        self.eslint = ESLintAPI(self)
        self.diff = DiffAPI(self)
        self.csv_parser = CsvParserAPI(self) # csv-parser
        self.mermaid_cli = MermaidCliAPI(self) # mermaid-cli
        self.pdfkit = PDFKitAPI(self)
        self.pillow = PillowAPI(self)

    def _request(self, method, path, params=None, data=None, json_data=None, files=None):
        """
        Internal method to make requests to the API.
        :param method: The HTTP method (e.g., 'GET', 'POST').
        :param path: The API endpoint path.
        :param params: Query parameters for GET requests.
        :param data: Form data for POST requests.
        :param json_data: JSON body for POST/PUT requests.
        :param files: Files for multipart/form-data uploads.
        :return: The response from the API (parsed JSON or raw content).
        :raises: requests.exceptions.HTTPError for bad responses (4xx or 5xx).
        """
        url = f'{self.base_url}{path}'
        headers = self.session.headers.copy()

        if json_data and files:
            raise ValueError("Cannot provide both json_data and files.")
        
        if files:
            # For multipart/form-data, let requests set the Content-Type
            if 'Content-Type' in headers:
                del headers['Content-Type']
        elif not data and not json_data:
             # For GET or POST with no body, ensure Content-Type is not set if not needed
            if method.upper() == 'GET' and 'Content-Type' in headers:
                 del headers['Content-Type']
            elif method.upper() == 'POST' and not data and not json_data and 'Content-Type' in headers:
                 # Keep Content-Type if it was explicitly set for an empty POST body, otherwise remove
                 pass # Or consider removing if it's the default application/json

        try:
            response = self.session.request(
                method,
                url,
                params=params,
                data=data,
                json=json_data,
                files=files,
                headers=headers
            )
            response.raise_for_status() # Raises HTTPError for bad responses (4xx or 5xx)

            content_type = response.headers.get('content-type', '').lower()
            if 'application/json' in content_type:
                return response.json()
            # For image, pdf, or other binary types, return raw content
            elif any(ct in content_type for ct in ['image', 'application/pdf']):
                return response.content
            return response.text # Fallback for other text-based types
        except requests.exceptions.RequestException as e:
            # Log or handle more specific exceptions as needed
            print(f"Request to {method} {url} failed: {e}")
            if hasattr(e, 'response') and e.response is not None:
                print(f"Response content: {e.response.text}")
            raise

class MarkedAPI:
    def __init__(self, client):
        self.client = client

    def render(self, markdown_text):
        """Converts Markdown text to HTML."""
        return self.client._request('POST', '/marked/render', json_data={'markdown': markdown_text})

class BeautifulSoupAPI:
    def __init__(self, client):
        self.client = client

    def parse(self, html, parser=None): # Changed html_content to html, added parser
        """Parses HTML content."""
        payload = {'html': html}
        if parser:
            payload['parser'] = parser
        return self.client._request('POST', '/beautifulsoup/parse', json_data=payload)

    def extract(self, html, selector, **kwargs): # Changed html_content to html
        """Extracts specific elements from HTML."""
        payload = {'html': html, 'selector': selector, **kwargs}
        return self.client._request('POST', '/beautifulsoup/extract', json_data=payload)

    def links(self, html, base_url=None, parser=None): # Changed html_content to html, added base_url and parser
        """Extracts all links from HTML."""
        payload = {'html': html}
        if base_url:
            payload['base_url'] = base_url
        if parser:
            payload['parser'] = parser
        return self.client._request('POST', '/beautifulsoup/links', json_data=payload)

    def images(self, html, base_url=None, parser=None): # Changed html_content to html, added base_url and parser
        """Extracts all images from HTML."""
        payload = {'html': html}
        if base_url:
            payload['base_url'] = base_url
        if parser:
            payload['parser'] = parser
        return self.client._request('POST', '/beautifulsoup/images', json_data=payload)

    def clean(self, html, remove_tags=None, keep_only=None, remove_comments=None, parser=None): # Changed html_content to html, added other params
        """Cleans HTML content."""
        payload = {'html': html}
        if remove_tags is not None:
            payload['remove_tags'] = remove_tags
        if keep_only is not None:
            payload['keep_only'] = keep_only
        if remove_comments is not None:
            payload['remove_comments'] = remove_comments
        if parser:
            payload['parser'] = parser
        return self.client._request('POST', '/beautifulsoup/clean', json_data=payload)

    def fetch(self, url, selector=None, parser=None): # Changed url_to_fetch to url, added selector and parser
        """Fetches a webpage and parses its content."""
        payload = {'url': url}
        if selector:
            payload['selector'] = selector
        if parser:
            payload['parser'] = parser
        return self.client._request('POST', '/beautifulsoup/fetch', json_data=payload)

    def health(self):
        """Health check for BeautifulSoup API."""
        return self.client._request('GET', '/beautifulsoup/health')

class PrettierAPI:
    def __init__(self, client):
        self.client = client

    def format(self, code, parser, options=None):
        """Formats code using Prettier."""
        payload = {'code': code, 'parser': parser}
        if options: payload['options'] = options
        return self.client._request('POST', '/prettier/format', json_data=payload)

    def check(self, code, parser, options=None):
        """Checks code formatting using Prettier."""
        payload = {'code': code, 'parser': parser}
        if options: payload['options'] = options
        return self.client._request('POST', '/prettier/check', json_data=payload)
    
    def batch(self, files, parser, options=None):
        """Batch format multiple code snippets."""
        # Assuming files is a list of dicts like [{ "name": "file.js", "code": "..." }]
        payload = {'files': files, 'parser': parser}
        if options: payload['options'] = options
        return self.client._request('POST', '/prettier/batch', json_data=payload)

    def get_parsers(self):
        """Gets available Prettier parsers."""
        return self.client._request('GET', '/prettier/parsers')

    def get_options(self):
        """Gets Prettier configuration options."""
        return self.client._request('GET', '/prettier/options')

    def health(self):
        """Health check for Prettier API."""
        return self.client._request('GET', '/prettier/health')

    def get_info(self):
        """Get Prettier API information."""
        return self.client._request('GET', '/prettier/info')

class PygmentsAPI:
    def __init__(self, client):
        self.client = client

    def highlight(self, code, language, **kwargs):
        """Highlights code syntax using Pygments."""
        payload = {'code': code, 'language': language, **kwargs}
        return self.client._request('POST', '/pygments/highlight', json_data=payload)

class PythonQRCodeAPI:
    def __init__(self, client):
        self.client = client

    def generate_qrcode(self, data, **kwargs):
        """Generates a QR code image. Returns image bytes."""
        payload = {'data': data, **kwargs}
        return self.client._request('POST', '/python-qrcode/generate-qrcode', json_data=payload)

class SanitizeHtmlAPI:
    def __init__(self, client):
        self.client = client

    def sanitize(self, html_content, options=None):
        """Sanitizes HTML content to prevent XSS attacks."""
        payload = {'html_content': html_content} # Changed 'html' to 'html_content'
        if options: payload['options'] = options
        return self.client._request('POST', '/sanitize-html/sanitize-html', json_data=payload)

class AjvAPI:
    def __init__(self, client):
        self.client = client

    def validate(self, schema, data_to_validate):
        """Validates JSON data against a schema using Ajv."""
        payload = {'schema': schema, 'data': data_to_validate}
        return self.client._request('POST', '/ajv/validate', json_data=payload)

class ESLintAPI:
    def __init__(self, client):
        self.client = client

    def lint(self, code, language='javascript', fix=False, **kwargs):
        """Lints JavaScript/TypeScript code using ESLint."""
        payload = {'code': code, 'language': language, 'fix': fix, **kwargs}
        # Removed redundant 'rules' parameter, pass rules directly in kwargs if needed by API
        return self.client._request('POST', '/eslint/lint', json_data=payload)

class DiffAPI:
    def __init__(self, client):
        self.client = client

    def compare(self, text1, text2, type='lines', outputFormat='json', **kwargs):
        """Compares two texts and returns the differences."""
        payload = {'text1': text1, 'text2': text2, 'type': type, 'outputFormat': outputFormat, **kwargs}
        return self.client._request('POST', '/diff/compare', json_data=payload)

class CsvParserAPI:
    def __init__(self, client):
        self.client = client

    def parse(self, csv_data, options=None, **kwargs):
        """Parses CSV data into JSON."""
        payload = {'csv_data': csv_data}
        if options is not None:  # Ensure options is only added if provided
            payload['options'] = options
        payload.update(kwargs) # Merge any additional kwargs
        return self.client._request('POST', '/csv-parser/parse', json_data=payload)

class MermaidCliAPI:
    def __init__(self, client):
        self.client = client

    def generate_diagram(self, definition, format='svg', **kwargs):
        """Generates a diagram from Mermaid text definition. Returns image bytes."""
        payload = {'definition': definition, 'format': format, **kwargs}
        # Initial request to generate the diagram and get the file path
        json_response = self.client._request('POST', '/mermaid-cli/generate-diagram', json_data=payload)
        
        if isinstance(json_response, dict) and 'filePath' in json_response:
            file_path = json_response['filePath']
            # Second request to fetch the actual diagram content
            # The _request method should handle returning bytes based on content type
            return self.client._request('GET', file_path) 
        else:
            # Handle cases where the expected filePath is not in the response
            raise Exception(f"Failed to retrieve diagram file path from MermaidCliAPI: {json_response}")

class PDFKitAPI:
    def __init__(self, client):
        self.client = client

    def generate(self, text_content, **kwargs):
        """Generates a PDF document using PDFKit. Returns PDF bytes."""
        payload = {'text_content': text_content, **kwargs}
        # Initial request to generate the PDF and get the file path
        json_response = self.client._request('POST', '/pdfkit/generate-pdf', json_data=payload)

        if isinstance(json_response, dict) and 'filePath' in json_response:
            file_path = json_response['filePath']
            # Second request to fetch the actual PDF content
            # The _request method should handle returning bytes based on content type
            return self.client._request('GET', file_path)
        else:
            # Handle cases where the expected filePath is not in the response
            raise Exception(f"Failed to retrieve PDF file path from PDFKitAPI: {json_response}")

class PillowAPI:
    def __init__(self, client):
        self.client = client

    def process(self, file_to_upload, operations, output_format='PNG', **kwargs):
        """
        Processes and edits images using Pillow. Returns image bytes.
        :param file_to_upload: Bytes of the image file.
        :param operations: List of operation strings (e.g., ["resize:200,200", "grayscale"]).
        :param output_format: Desired output format (e.g., 'PNG', 'JPEG').
        """
        if not isinstance(file_to_upload, bytes):
            raise ValueError("file_to_upload must be bytes.")
        
        files = {'file': ('image_file', file_to_upload)} 
        
        # API expects 'operations' as multiple form fields, not a single JSON string.
        # And other kwargs also as form fields.
        form_data = [('output_format', output_format)]
        for op in operations:
            form_data.append(('operations', op))
        
        for key, value in kwargs.items():
            form_data.append((key, str(value))) # Ensure all values are strings for form data

        # The _request method should handle 'data' being a list of tuples for multipart/form-data
        # when 'files' is also present. 'requests' library supports this.
        return self.client._request('POST', '/pillow/process-image', data=form_data, files=files)

if __name__ == '__main__':
    # Example Usage (Illustrative - requires running AllBeApi services)
    # client = AllBeApi()

    # try:
        # Markdown to HTML
        # html_output = client.marked.render("# Hello Python SDK")
        # print("Marked Output:", html_output)

        # # QR Code Generation (returns bytes)
        # qr_image_bytes = client.python_qrcode.generate_qrcode("https://res.allbeapi.top")
        # with open("example_qrcode.png", "wb") as f:
        #     f.write(qr_image_bytes)
        # print("QR Code saved to example_qrcode.png")

        # # Prettier format (example)
        # formatted_code = client.prettier.format("const foo =    1", "babel")
        # print("Prettier Output:", formatted_code)

        # # BeautifulSoup parse (example)
        # parsed_html = client.beautifulsoup.parse("<html><body><h1>Hi</h1></body></html>")
        # print("BeautifulSoup Parsed:", parsed_html)
        
        # # Pillow API (example with URL)
        # # Note: This is a placeholder URL and operation. The actual API might require specific image URLs and operations.
        # # processed_image_bytes = client.pillow.process(image_url_or_data="https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png", operation="resize", width=100, height=50)
        # # with open("processed_image.png", "wb") as f:
        # #     f.write(processed_image_bytes)
        # # print("Processed image saved to processed_image.png")

    # except requests.exceptions.HTTPError as e:
    #     print(f"API Error: {e.response.status_code} - {e.response.text}")
    # except Exception as e:
    #     print(f"An unexpected error occurred: {e}")
    pass

File: SDK/JavaScript/allbeapi.js
/**
 * AllBeAPI JavaScript SDK - Minimal Utility Function Set
 * Base URL: https://res.allbeapi.top
 *
 * Rapid development toolkit for prototyping and experimentation.
 * Access ready-to-use functions without installing heavy dependencies.
 */

class AllBeApi {
    /**
     * Initializes a new instance of the AllBeApi client.
     * @param {string} [baseUrl='https://res.allbeapi.top'] - The base URL for the API.
     */
    constructor(baseUrl = 'https://res.allbeapi.top') {
        this.baseUrl = baseUrl;

        this.marked = new MarkedAPI(this);
        this.beautifulsoup = new BeautifulSoupAPI(this);
        this.prettier = new PrettierAPI(this);
        this.pygments = new PygmentsAPI(this);
        this.pythonQrcode = new PythonQRCodeAPI(this); // python-qrcode -> pythonQrcode
        this.sanitizeHtml = new SanitizeHtmlAPI(this); // sanitize-html -> sanitizeHtml
        this.ajv = new AjvAPI(this);
        this.eslint = new ESLintAPI(this);
        this.diff = new DiffAPI(this);
        this.csvParser = new CsvParserAPI(this); // csv-parser -> csvParser
        this.mermaidCli = new MermaidCliAPI(this); // mermaid-cli -> mermaidCli
        this.pdfkit = new PDFKitAPI(this); // pdfkit -> pdfkit (using PDFKitAPI for consistency)
        this.pillow = new PillowAPI(this);
    }

    /**
     * Internal method to make requests to the API.
     * @param {string} method - The HTTP method (e.g., 'GET', 'POST').
     * @param {string} path - The API endpoint path.
     * @param {object} [queryParams=null] - Query parameters for GET requests.
     * @param {object} [body=null] - The request body for POST/PUT requests.
     * @returns {Promise<any>} - The response from the API.
     * @private
     */
    async _request(method, path, queryParams = null, body = null) {
        let url = `${this.baseUrl}${path}`;
        const options = {
            method: method,
            headers: {}
        };

        if (queryParams) {
            const params = new URLSearchParams();
            for (const key in queryParams) {
                if (queryParams.hasOwnProperty(key) && queryParams[key] !== undefined) {
                    params.append(key, queryParams[key]);
                }
            }
            if (params.toString()) {
                 url = `${url}?${params}`;
            }
        }

        if (body) {
            if (body instanceof FormData) {
                options.body = body; // For file uploads or form data
            } else {
                options.headers['Content-Type'] = 'application/json';
                options.body = JSON.stringify(body);
            }
        }

        const response = await fetch(url, options);

        if (!response.ok) {
            const errorData = await response.text();
            throw new Error(`API request to ${method} ${url} failed with status ${response.status}: ${errorData}`);
        }

        const contentType = response.headers.get('content-type');

        // Handle non-JSON responses based on known endpoints
        if (path.includes('/python-qrcode/generate-qrcode') ||
            path.includes('/mermaid-cli/generate-diagram') ||
            path.includes('/pillow/process-image')) {
            return response.blob(); // Expected to be image/*
        } else if (path.includes('/pdfkit/generate-pdf')) {
            return response.blob(); // Expected to be application/pdf
        } else if (contentType && contentType.includes('application/json')) {
            return response.json();
        }
        
        return response.text(); // Fallback for other types or if content-type is missing
    }
}

class MarkedAPI {
    constructor(client) { this.client = client; }
    /**
     * 将 Markdown 文本转换为 HTML (Converts Markdown text to HTML)
     * @param {string} markdown - The Markdown content.
     * @returns {Promise<object>}
     */
    async render(markdown) {
        return this.client._request('POST', '/marked/render', null, { markdown });
    }
}

class BeautifulSoupAPI {
    constructor(client) { this.client = client; }
    /**
     * 解析HTML内容 (Parses HTML content)
     * @param {string} html - The HTML content to parse.
     * @param {string} [parser='html.parser'] - The parser to use.
     * @returns {Promise<object>}
     */
    async parse(html, parser = 'html.parser') {
        return this.client._request('POST', '/beautifulsoup/parse', null, { html, parser });
    }
    /**
     * 提取特定元素 (Extracts specific elements from HTML)
     * @param {string} html - The HTML content.
     * @param {string} selector - CSS selector to extract elements.
     * @param {string} [attribute=null] - Specific attribute to extract from elements.
     * @param {string} [parser='html.parser'] - The parser to use.
     * @returns {Promise<object>}
     */
    async extract(html, selector, attribute = null, parser = 'html.parser') {
        const payload = { html, selector, parser };
        if (attribute) {
            payload.attribute = attribute;
        }
        return this.client._request('POST', '/beautifulsoup/extract', null, payload);
    }
    /**
     * 提取所有链接 (Extracts all links from HTML)
     * @param {string} html - The HTML content.
     * @param {string} [base_url=null] - Base URL to resolve relative links.
     * @param {string} [parser='html.parser'] - The parser to use.
     * @returns {Promise<object>}
     */
    async links(html, base_url = null, parser = 'html.parser') {
        const payload = { html, parser };
        if (base_url) {
            payload.base_url = base_url;
        }
        return this.client._request('POST', '/beautifulsoup/links', null, payload);
    }
    /**
     * 提取所有图片 (Extracts all images from HTML)
     * @param {string} html - The HTML content.
     * @param {string} [base_url=null] - Base URL to resolve relative image paths.
     * @param {string} [parser='html.parser'] - The parser to use.
     * @returns {Promise<object>}
     */
    async images(html, base_url = null, parser = 'html.parser') {
        const payload = { html, parser };
        if (base_url) {
            payload.base_url = base_url;
        }
        return this.client._request('POST', '/beautifulsoup/images', null, payload);
    }
    /**
     * 清理HTML内容 (Cleans HTML content)
     * @param {string} html - The HTML content to clean.
     * @param {string[]} [remove_tags=[]] - List of tags to remove.
     * @param {string[]} [keep_only=[]] - List of tags to keep, removing others.
     * @param {boolean} [remove_comments=false] - Whether to remove comments.
     * @param {string} [parser='html.parser'] - The parser to use.
     * @returns {Promise<object>}
     */
    async clean(html, remove_tags = [], keep_only = [], remove_comments = false, parser = 'html.parser') {
        return this.client._request('POST', '/beautifulsoup/clean', null, { html, remove_tags, keep_only, remove_comments, parser });
    }
    /**
     * 获取网页并解析 (Fetches a webpage and parses its content)
     * @param {string} url - The URL of the webpage to fetch and parse.
     * @param {string} [parser='html.parser'] - The parser to use.
     * @returns {Promise<object>}
     */
    async fetch(url, parser = 'html.parser') {
        return this.client._request('POST', '/beautifulsoup/fetch', null, { url, parser });
    }
    /**
     * 健康检查 (Health check for BeautifulSoup API)
     * @returns {Promise<object>}
     */
    async health() {
        return this.client._request('GET', '/beautifulsoup/health');
    }
}

class PrettierAPI {
    constructor(client) { this.client = client; }
    /**
     * 格式化代码 (Formats code using Prettier)
     * @param {string} code - The code to format.
     * @param {string} parser - The parser to use (e.g., 'babel', 'typescript').
     * @param {object} [options] - Prettier options.
     * @returns {Promise<object>}
     */
    async format(code, parser, options = {}) {
        return this.client._request('POST', '/prettier/format', null, { code, parser, options });
    }
    /**
     * 检查代码格式 (Checks code formatting using Prettier)
     * @param {string} code - The code to check.
     * @param {string} parser - The parser to use.
     * @param {object} [options] - Prettier options.
     * @returns {Promise<object>}
     */
    async check(code, parser, options = {}) {
        return this.client._request('POST', '/prettier/check', null, { code, parser, options });
    }
    /**
     * 批量格式化 (Batch format multiple code snippets)
     * @param {Array<object>} files - Array of file objects, e.g., [{ name: 'file.js', code: '...' }].
     * @param {string} parser - The parser to use for all files.
     * @param {object} [options] - Prettier options.
     * @returns {Promise<object>}
     */
    async batch(files, parser, options = {}) { // Assuming structure based on typical batch operations
        // files should be an array of objects, e.g., [{ code: '...', language: 'javascript' (or parser: 'babel'), options: {} }]
        return this.client._request('POST', '/prettier/batch', null, { files, parser, options });
    }
    /**
     * 获取支持的解析器 (Gets available Prettier parsers)
     * @returns {Promise<object>}
     */
    async getParsers() {
        return this.client._request('GET', '/prettier/parsers');
    }
    /**
     * 获取配置选项 (Gets Prettier configuration options)
     * @returns {Promise<object>}
     */
    async getOptions() {
        return this.client._request('GET', '/prettier/options');
    }
    /**
     * 健康检查 (Health check for Prettier API)
     * @returns {Promise<object>}
     */
    async health() {
        return this.client._request('GET', '/prettier/health');
    }
    /**
     * API信息 (Get Prettier API information)
     * @returns {Promise<object>}
     */
    async getInfo() {
        return this.client._request('GET', '/prettier/info');
    }
}

class PygmentsAPI {
    constructor(client) { this.client = client; }
    /**
     * 对代码进行语法高亮 (Highlights code syntax using Pygments)
     * @param {string} code - The code to highlight.
     * @param {string} language - The language of the code.
     * @param {object} [options] - Additional Pygments options (e.g., formatter, style).
     * @returns {Promise<object>}
     */
    async highlight(code, language, style = 'default', formatter = 'html', options = {}) {
        return this.client._request('POST', '/pygments/highlight', null, { code, language, style, formatter, ...options });
    }
}

class PythonQRCodeAPI {
    constructor(client) { this.client = client; }
    /**
     * 生成二维码图像 (Generates a QR code image)
     * @param {string} data - The data to encode in the QR code.
     * @param {object} [options] - Options like size, box_size, border, fill_color, back_color.
     * @returns {Promise<Blob>} - The QR code image as a Blob.
     */
    async generateQrcode(data, options = {}) {
        return this.client._request('POST', '/python-qrcode/generate-qrcode', null, { data, ...options });
    }
}

class SanitizeHtmlAPI {
    constructor(client) { this.client = client; }
    /**
     * 清理HTML内容，防止XSS攻击 (Sanitizes HTML content to prevent XSS attacks)
     * @param {string} html_content - The HTML content to sanitize.
     * @param {object} [options] - Sanitization options.
     * @returns {Promise<object>}
     */
    async sanitize(html_content, options = {}) {
        // API endpoint is /sanitize-html/sanitize-html and expects 'html_content'
        return this.client._request('POST', '/sanitize-html/sanitize-html', null, { html_content, options });
    }
}

class AjvAPI {
    constructor(client) { this.client = client; }
    /**
     * 验证JSON数据是否符合Schema (Validates JSON data against a schema using Ajv)
     * @param {object} schema - The JSON schema.
     * @param {object} data - The JSON data to validate.
     * @returns {Promise<object>}
     */
    async validate(schema, data) {
        return this.client._request('POST', '/ajv/validate', null, { schema, data });
    }
}

class ESLintAPI {
    constructor(client) { this.client = client; }
    /**
     * 对JavaScript/TypeScript代码进行静态分析 (Lints JavaScript/TypeScript code using ESLint)
     * @param {string} code - The code to lint.
     * @param {string} language - 'javascript' or 'typescript'.
     * @param {boolean} [fix=false] - Whether to attempt to fix linting issues.
     * @param {object} [eslintOptions] - Additional ESLint options (currently not directly used by API, but kept for future).
     * @returns {Promise<object>}
     */
    async lint(code, language, fix = false, eslintOptions = {}) {
        // API endpoint is /eslint/lint and expects 'code', 'language', 'fix'
        return this.client._request('POST', '/eslint/lint', null, { code, language, fix });
    }
}

class DiffAPI {
    constructor(client) { this.client = client; }
    /**
     * 比较两个文本的差异 (Compares two texts and returns the differences)
     * @param {string} text1 - The first text.
     * @param {string} text2 - The second text.
     * @param {object} [options] - Diffing options.
     * @returns {Promise<object>}
     */
    async compare(text1, text2, options = {}) {
        return this.client._request('POST', '/diff/compare', null, { text1, text2, ...options });
    }
}

class CsvParserAPI {
    constructor(client) { this.client = client; }
    /**
     * 将CSV转换为JSON (Parses CSV data into JSON)
     * @param {string} csvData - The CSV data as a string.
     * @param {object} [options] - CSV parsing options (e.g., delimiter, headers).
     * @returns {Promise<object>}
     */
    async parse(csvData, options = {}) {
        return this.client._request('POST', '/csv-parser/parse', null, { csv_data: csvData, options: options });
    }
}

class MermaidCliAPI {
    constructor(client) { this.client = client; }
    /**
     * 从文本定义生成图表 (Generates a diagram from Mermaid text definition)
     * @param {string} definition - The Mermaid diagram definition.
     * @param {string} [format='svg'] - Output format ('svg' or 'png').
     * @param {object} [options] - Additional options (currently not directly used by API, but kept for future).
     * @returns {Promise<Blob>} - The diagram image as a Blob.
     */
    async generateDiagram(definition, format = 'svg', options = {}) {
        // API returns a JSON with filePath, then the file is served from that path.
        const response = await this.client._request('POST', '/mermaid-cli/generate-diagram', null, { definition, format, ...options });
        if (response && response.filePath) {
            // The _request method should handle fetching the blob directly if the path is absolute
            // or if it's a relative path from the base URL.
            // Assuming filePath is relative to the base API URL.
            return this.client._request('GET', response.filePath);
        }
        throw new Error('Failed to retrieve diagram file path from MermaidCliAPI');
    }
}

class PDFKitAPI { 
    constructor(client) { this.client = client; }
    /**
     * 生成PDF文档 (Generates a PDF document using PDFKit)
     * @param {string} text_content - The text content for the PDF.
     * @param {object} [options] - PDF generation options (currently not directly used by API, but kept for future).
     * @returns {Promise<Blob>} - The PDF document as a Blob.
     */
    async generate(text_content, options = {}) {
        // API returns a JSON with filePath, then the file is served from that path.
        const response = await this.client._request('POST', '/pdfkit/generate-pdf', null, { text_content, ...options });
        if (response && response.filePath) {
            return this.client._request('GET', response.filePath);
        }
        throw new Error('Failed to retrieve PDF file path from PDFKitAPI');
    }
}

class PillowAPI {
    constructor(client) { this.client = client; }
    /**
     * 处理和编辑图像 (Processes and edits images using Pillow)
     * @param {File} file - The image file to process.
     * @param {string[]} operations - Array of operation strings (e.g., ["resize:200,200", "grayscale"]).
     * @param {string} [output_format='PNG'] - The desired output format (e.g., 'PNG', 'JPEG').
     * @returns {Promise<Blob>} - The processed image as a Blob.
     */
    async process(file, operations, output_format = 'PNG') {
        const formData = new FormData();
        formData.append('file', file);
        // API expects 'operations' as multiple form fields.
        operations.forEach(op => formData.append('operations', op));
        formData.append('output_format', output_format);
        // API directly returns the processed image blob.
        return this.client._request('POST', '/pillow/process-image', null, formData);
    }
}

// Export the main class for Node.js environments or as a global for browsers.
if (typeof module !== 'undefined' && module.exports) {
    module.exports = AllBeApi;
    module.exports.default = AllBeApi; // For ES6 default import
} else if (typeof window !== 'undefined') {
    window.AllBeApi = AllBeApi;
}
